include/libvfio-user.h

```diff
@@ -268,33 +246,106 @@ typedef enum {
 } vfu_pci_type_t;
 
 /**
- * Setup PCI configuration space header data. This function must be called only
- * once per libvfio-user context.
+ * Initialize the context for a PCI device. Returns an initialized buffer for a
+ * type 0 PCI configuration space of the needed size.
+ *
+ * This buffer consists of the standard 64-byte PCI header and the remaining
+ * space depending on @pci_type, either PCI_CFG_SPACE_SIZE or
+ * PCI_CFG_SPACE_EXP_SIZE.
  *
  * @vfu_ctx: the libvfio-user context
- * @id: Device and vendor ID
- * @ss: Subsystem vendor and device ID
- * @cc: Class code
  * @pci_type: PCI type (convention PCI, PCI-X mode 1, PCI-X mode2, PCI-Express)
  * @revision: PCI/PCI-X/PCIe revision
+ */
+vfu_pci_hdr_t *
+vfu_pci_init(vfu_ctx_t *vfu_ctx, vfu_pci_type_t pci_type, int revision UNUSED);
+
+/*
+ * FIXME
+ */
+void
+vfu_pci_set_id(vfu_ctx_t *vfu_ctx, uint16_t vid, uint16_t did,
```

Comment:
We can configure these via a single function, we should look into the PCI spec and see what can be configured by the device.

Reply:
I've split them up because it looks way clearer to me than one function with loads of params and little context as to what is what.

Reply:
> Instead why not have a struct(e.g vfu_pci_std_hdr) around the standard header bits which we want user to specify. It should be unlikely that this struct will change.
> Having multiple functions is little inconvenient to users from my POV.

A struct similar to `lm_pci_info_t`? The one we had and removed? Not sure it's worth to have a struct for just 3 parameters IMO.